### ISSUE TITLE
Error when npm i -D naive-ui

### ISSUE THREAD

**Issue by LehaoLin** [TuSimple/naive-ui#2158]
### TuSimple/naive-ui version (版本)
2.24.0

### Vue version (Vue 版本)
3.2.26

### Browser and its version (浏览器及其版本)
Chrome

### System and its version (系统及其版本)
MacOS

### Node version (Node 版本)
v16.13.1

### Reappearance link (重现链接)
www.naiveui.com

### Reappearance steps (重现步骤)
npm i -D naive-ui

### Expected results (期望的结果)
Fix the bug

### Actual results (实际的结果)
npm ERR! command failed
npm ERR! command sh -c pnpm run gen-component-declaration
npm ERR! sh: pnpm: command not found

**Comment by woodhull** [comment]
I also have this issue, fixed by downgrading to `2.23.1`

**Comment by sagaban** [comment]
I am having the same issue with `yarn add -D naive-ui vfonts`. I tried installing `pnpm` and `esno` with no luck. I would like to use a new introduced functionality in 2.24 version, downgrading won't work for me.

**Comment by littleCareless** [comment]
I tried installing `esno` globally and it worked a little, but then a new error appeared.
![ACBB5304-EF60-4535-BD24-18DD2F49816F](https://user-images.githubusercontent.com/47601009/149045063-c9623d3d-e821-406c-b767-e85d15692abd.png)

**Comment by Lanseria** [comment]
升级 `yarn upgrade-interactive` 出现
```shell
Error: C:\Users\WS\Documents\Code\i-share-clue\pc\node_modules\naive-ui: Command failed.
Exit code: 1
Command: pnpm run gen-component-declaration
Arguments:
Directory: C:\Users\WS\Documents\Code\i-share-clue\pc\node_modules\naive-ui
Output:
'pnpm' �����ڲ����ⲿ���Ҳ���ǿ����еĳ���
���������ļ���
    at ProcessTermError.ExtendableBuiltin (C:\Users\WS\AppData\Roaming\npm\node_modules\yarn\lib\cli.js:721:66)

```

**Comment by 07akioni** [comment]
closed in 55831d9540703bbf49ce03ab93a0aa48ce33c282
